src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java

```diff
@@ -0,0 +1,246 @@
+        PowerMockito.doReturn(observable).when(AzurePublicIPAddressSDK.class, "associateNetworkSecurityGroupAsync",
+                Mockito.eq(networkInterface), Mockito.eq(creatable));
+
+        Mockito.doNothing().when(this.operation).subscribeUpdateNetworkInterface(Mockito.eq(observable));
+
+        // exercise
+        this.operation.handleSecurityIssues(azure, instanceId, networkInterface);
+
+        // verify
+        Mockito.verify(azure, Mockito.times(TestUtils.RUN_ONCE)).publicIPAddresses();
+        Mockito.verify(publicIPAddresses, Mockito.times(TestUtils.RUN_ONCE))
+                .getByResourceGroup(Mockito.eq(this.resourceGroupName), Mockito.eq(instanceId));
+
+        PowerMockito.verifyStatic(AzurePublicIPAddressSDK.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzurePublicIPAddressSDK.buildNetworkSecurityGroupCreatable(Mockito.eq(azure), Mockito.eq(publicIPAddress));
+
+        PowerMockito.verifyStatic(AzurePublicIPAddressSDK.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzurePublicIPAddressSDK.associateNetworkSecurityGroupAsync(Mockito.eq(networkInterface), Mockito.eq(creatable));
+
+        Mockito.verify(this.operation, Mockito.times(TestUtils.RUN_ONCE))
+                .subscribeUpdateNetworkInterface(Mockito.eq(observable));
+    }
+
+    // test case: When calling the subscribeUpdateNetworkInterface method and the
+    // observable executes without any error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeUpdateNetworkInterfaceSuccessfully() {
+        // set up
+        Observable<NetworkInterface> observable = Observable.defer(() -> {
+            NetworkInterface networkInterface = Mockito.mock(NetworkInterface.class);
+            return Observable.just(networkInterface);
+        });
+
+        // exercise
+        this.operation.subscribeUpdateNetworkInterface(observable);
+
+        // verify
+        this.loggerAssert.assertEqualsInOrder(Level.INFO, Messages.Info.END_UPDATE_NIC_ASYNC_BEHAVIOUR);
+    }
+
+    // test case: When calling the subscribeUpdateNetworkInterface method and the
+    // observable executes with an error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeUpdateNetworkInterfaceFail() {
+        // set up
+        Observable observable = AzureTestUtils.createSimpleObservableFail();
+
+        // exercise
+        this.operation.subscribeUpdateNetworkInterface(observable);
+
+        // verify
+        this.loggerAssert.assertEqualsInOrder(Level.ERROR, Messages.Error.ERROR_UPDATE_NIC_ASYNC_BEHAVIOUR);
+    }
+
+    // test case: When calling the subscribeDisassociateAndDeleteResources method
+    // and the observable executes without any error, it must verify than returns
+    // the right logs.
+    @Test
+    public void testSubscribeDisassociateAndDeleteResourcesSuccessfully() {
+        // set up
+        Completable completable = Completable.complete();
```

Comment:
You should use a method that belongs to AzureTestUtils.
[Link](https://github.com/fogbow/resource-allocation-service/blob/develop/src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/AzureTestUtils.java#L50)
Look at the whole code about this same issue.

Reply:
I'll fix It and the other occurrences.